doc/making-a-visualizer.md

```diff
@@ -14,55 +20,64 @@ the following data and methods:
+   size is already set up to comprise the available space for visualization.
+   However, it should not yet perform any visualization.
+5. `validate()`: must return `ValidationStatus` object that indicates to the
    engine that the visualizer is valid. The engine will call `validate` before
    it calls `initialize` and will only proceed if the `isValid` property of
    the `ValidationStatus` object is `true`. Otherwise it will display the
    error. Generally speaking, the framework takes care of these bookkeeping
    details and you can simply implement the `checkParameters()` method that
    just has to examine if parameter values are sensible and return a
    ValidationStatus accordingly.
-7. `setup()`: a method that acts on the p5 sketch to set up the canvas for
-   drawing. Called just prior to draw.
-8. `draw()`: the method that does the bulk of the visualizer's work. This is
-   where you can get creative. Generally acts on the sketch p5 object and
-   actually draws the visualization. For details on this, refer to the p5
-   documentation.
+6. `show()`: Begin (or resume) displaying the visualization.
+7. `stop()`: Pause displaying the visualization (but don't erase any
+   visualization produced so far or otherwise clean up the visualizer).
+8. `dispose()`: Throw out the visualization, release its resources, remove its
+   injected DOM elements, etc. Note that after this call, the visualizer must
+   support `inhabit()` being called again, perhaps with a different div, to
+   re-initialize the visualization.
 
 ### The easy way
 
-The simplest and fastest way to set up your visualizer is to extend the
-`VisualizerDefault` class. This guarantees that you will implement the
-interface, however you will still need to supply your collection of `params`,
-and provide the details of your `checkParameters`, `initialize`, `setup`, and
-`draw` functions. The default class provides a rough template for the best way
-to structure a visualizer.
-
-The provided example visualizers are good starting points for how to extend
-the default class.
+The simplest and fastest way to set up your visualizer is to use the `p5.js`
+library to create its graphical output. You can do this by extending the
+`P5Visualizer` base class. This approach guarantees that you will implement
+the necessary interface. However, you will still need to supply your
+collection of `params`, and provide the details of your `checkParameters`. You
+can optionally implement the `inhabit` function if you have initialization you
+want performed whenever the visualizer is inserted onto the page, but be
+certain to call `super.inhabit` therein to get the standard P5Visualizer
+initialization as well.
+
+You shouldn't need to implement `show()`, `stop()`, or `dispose()`, but you
+can implement a `setup()` function as usual with [p5](https://p5js.org/learn/)
+if needed to set the stage of the graphics for your visualization. If you do,
```

Comment:
"set the stage of the graphics" is a big vague

Reply:
Hopefully it's better now?